### ISSUE TITLE
Sort out NuGet packages

### ISSUE THREAD

**Issue by alexeyzimarev** [AsynkronIT/protoactor-dotnet#501]
Right now we have packages for each project. However, it doesn't seem right since some packages cannot be used in isolation in reality. For example:

- To use ProtoActor you really need `Proto.Actor` and `Proto.Mailbox`
- For simple remote scenarios you'd need both `Proto.Remote` and `Proto.Router`

I suggest the following:

- We won't publish `Proto.Mailbox` as a package. The assembly will be a part of `Proto.Actor` package
- Same for `Proto.Router` - it will be a part of `Proto.Remote`

**Comment by Damian-P** [comment]
Proto.remote doesn't need the Proto.Router project. We can remove the reference in the project.

But the router logic could be added to the ActorSystem since it is accessible from everywhere now.

**Comment by alexeyzimarev** [comment]
The question is then why Mailbox and Router are in separate projects? Could the core project contain all it requires to form a usable package?

**Comment by rogeralsing** [comment]
>The question is then why Mailbox and Router are in separate projects? Could the core project contain all it requires to form a usable package?

I wrote the Go version of ProtoActor first. and the Go package structure, which is just folders, followed over to .NET

>Same for Proto.Router - it will be a part of Proto.Remote

Those two should not be merged, routers are not associated with remote.

Mailboxes can in theory be used in isolation, e.g. when building background logging features. not sure if that is worth separating for though.

Routers should likely be merged into core proto.actor package.

thoughts @cpx86 , @PotterDai  ?

personally, I do think there is some value of the separation as it keeps the core small, and does not allow actor and router code to be too intertwined.

**Comment by alexeyzimarev** [comment]
My point is that people should be able to install one or two packages only to get going. Even if projects are separated, we don't necessarily need to publish each project as a package. For example, the Proto.Actor package can contain the Mailbox assembly as well.

**Comment by alexeyzimarev** [comment]
Ok, I left the package structure as it was. Just ensured that everything gets published on NuGet, we were missing a couple of packages with the new pipeline.

**Comment by cpx86** [comment]
@rogeralsing Regarding Mailbox, I think the core mechanics of it (blocking/unblocking, bounded/unbounded queues) is reusable, but the way we designed the interface feels quite specific to protoactor. Might be interesting exploring making something more general-purpose from it, but in it's current shape I think it could be merged into Proto.Actor.
